pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -213,10 +188,325 @@ func diffDescription(diff DiffType) string {
+	alignedConnectivityConfig, alignedOtherConnectivityConfig *SubnetConfigConnectivity, myErr error) {
+	// 1. computes new set of external nodes (only type of nodes here) in cfg1 and cfg2
+	// does so by computing disjoint block between src+dst ipBlocks in cfg1 and in cfg2
+	// the new set of external nodes is determined based on them
+	connectivityIPBlist, err := subnetConfConnectivity.subnetConnectivity.getIPBlocksList()
+	if err != nil {
+		return nil, nil, err
+	}
+	otherIPBlist, err := otherConfConnectivity.subnetConnectivity.getIPBlocksList()
+	if err != nil {
+		return nil, nil, err
+	}
+	disjointIPblocks := common.DisjointIPBlocks(connectivityIPBlist, otherIPBlist)
+	// 2. copy configs and generates Nodes[] as per disjointIPblocks
+	alignedConfig, err := subnetConfConnectivity.config.copyConfig(disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	otherAlignedConfig, err := otherConfConnectivity.config.copyConfig(disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	// 3. resize connections as per the new Nodes[]
+	alignedConnectivity, err := subnetConfConnectivity.subnetConnectivity.alignConnectionsGivenIPBlists(
+		alignedConfig, disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	alignedOtherConnectivity, err := otherConfConnectivity.subnetConnectivity.alignConnectionsGivenIPBlists(
+		otherAlignedConfig, disjointIPblocks)
+	if err != nil {
+		return nil, nil, err
+	}
+	return &SubnetConfigConnectivity{alignedConfig, alignedConnectivity},
+		&SubnetConfigConnectivity{otherAlignedConfig, alignedOtherConnectivity}, myErr
+}
+
+func (subnetConnectivity *SubnetConnectivityMap) alignConnectionsGivenIPBlists(config *VPCConfig, disjointIPblocks []*common.IPBlock) (
+	alignedConnectivity SubnetConnectivityMap, err error) {
+	alignedConnectivitySrc, err := subnetConnectivity.actualAlignSrcOrDstGivenIPBlists(config, disjointIPblocks, true)
+	if err != nil {
+		return nil, err
+	}
+	alignedConnectivity, err = alignedConnectivitySrc.actualAlignSrcOrDstGivenIPBlists(config, disjointIPblocks, false)
+	return alignedConnectivity, err
+}
+
+// aligned config: copies from old config everything but nodes,
+// nodes are resized by disjointIPblocks
+func (c *VPCConfig) copyConfig(disjointIPblocks []*common.IPBlock) (alignedConfig *VPCConfig, err error) {
+	// copy config
+	alignedConfig = c
+	//  nodes - external addresses - are resized
+	alignedConfig.Nodes, err = resizeNodes(c.Nodes, disjointIPblocks)
+	return alignedConfig, err
+}
+
+func resizeNodes(oldNodes []Node, disjointIPblocks []*common.IPBlock) (newNodes []Node, err error) {
+	newNodes = []Node{}
+	//  range over old nodes and inside range over disjoint blocks
+	//  if a disjoint block is contained in an old oldNode - create external oldNode and add it
+	//  if no disjoint block is contained in an old oldNode - add the old oldNode as is
+	for _, oldNode := range oldNodes {
+		nodeIPBlock, err := common.NewIPBlock(oldNode.Cidr(), nil)
```

Comment:
In this stage oldNode can not be an internal node, since its only about subnets diff. So we have either external nodes of type Node or subnets of type NodeSets.

But you are right - once (soon) I'll extend this to support diff_all_endpoints the code here will change

Reply:
But the `oldNodes` is the config nodes, and it may contain internal nodes even in the analysis type is `all_subnets`.

Reply:
Right. Seems that what happened was that internal nodes were added as is by luck since `nodeIPBlock` was nil and `disjointContained` stayed false. Fixed.